apis/eks/v1beta1/types.go

```diff
@@ -0,0 +1,313 @@
+	// API server. For more information, see Amazon EKS Cluster Endpoint Access
+	// Control (https://docs.aws.amazon.com/eks/latest/userguide/cluster-endpoint.html)
+	// in the Amazon EKS User Guide.
+	// +optional
+	EndpointPublicAccess *bool `json:"endpointPublicAccess,omitempty"`
+
+	// The CIDR blocks that are allowed access to your cluster's public Kubernetes
+	// API server endpoint. Communication to the endpoint from addresses outside
+	// of the CIDR blocks that you specify is denied. The default value is 0.0.0.0/0.
+	// If you've disabled private endpoint access and you have worker nodes or AWS
+	// Fargate pods in the cluster, then ensure that you specify the necessary CIDR
+	// blocks. For more information, see Amazon EKS Cluster Endpoint Access Control
+	// (https://docs.aws.amazon.com/eks/latest/userguide/cluster-endpoint.html)
+	// in the Amazon EKS User Guide.
+	// +optional
+	PublicAccessCidrs []string `json:"publicAccessCidrs,omitempty"`
+
+	// Specify one or more security groups for the cross-account elastic network
+	// interfaces that Amazon EKS creates to use to allow communication between
+	// your worker nodes and the Kubernetes control plane. If you don't specify
+	// a security group, the default security group for your VPC is used.
+	// +optional
+	SecurityGroupIDs []string `json:"securityGroupIds,omitempty"`
+
+	// SecurityGroupIDRefs are references to SecurityGroups used to set
+	// the SecurityGroupIDs.
+	// +optional
+	SecurityGroupIDRefs []runtimev1alpha1.Reference `json:"securityGroupIDRefs,omitempty"`
+
+	// SecurityGroupIDSelector selects references to SecurityGroups used
+	// to set the SecurityGroupIDs.
+	// +optional
+	SecurityGroupIDSelector *runtimev1alpha1.Selector `json:"securityGroupIDSelector,omitempty"`
+
+	// Specify subnets for your Amazon EKS worker nodes. Amazon EKS creates cross-account
+	// elastic network interfaces in these subnets to allow communication between
+	// your worker nodes and the Kubernetes control plane.
+	// +optional
+	SubnetIDs []string `json:"subnetIds,omitempty"`
+
+	// SubnetIDRefs are references to Subnets used to set
+	// the SubnetIDs.
+	// +optional
+	SubnetIDRefs []runtimev1alpha1.Reference `json:"subnetIDRefs,omitempty"`
+
+	// SubnetIDSelector selects references to Subnets used
+	// to set the SubnetIDs.
+	// +optional
+	SubnetIDSelector *runtimev1alpha1.Selector `json:"subnetIDSelector,omitempty"`
+}
+
+// ClusterObservation is the observed state of a cluster.
+type ClusterObservation struct {
+	// The Amazon Resource Name (ARN) of the cluster.
+	Arn string `json:"arn,omitempty"`
+
+	// The Unix epoch timestamp in seconds for when the cluster was created.
+	CreatedAt *metav1.Time `json:"createdAt,omitempty"`
+
+	// The endpoint for your Kubernetes API server.
+	Endpoint string `json:"endpoint,omitempty"`
+
+	// The identity provider information for the cluster.
+	Identity Identity `json:"identity,omitempty"`
```

Comment:
Another field that I debated whether including in the status or not. It is not in our connection `Secret` currently so I left it.